File: starters/cs/MyTronBot.cs
using System;
using System.Collections.Generic;

class MyTronBot{

   public static string MakeMove()
   {
    int x = Map.MyX();
    int y = Map.MyY();
    List<string> validMoves = new List<string>();
    
    if (!Map.IsWall(x,y-1)) {
	    validMoves.Add("North");
	 }
	 if (!Map.IsWall(x+1,y)) {
	    validMoves.Add("East");
	 }
	 if (!Map.IsWall(x,y+1)) {
	    validMoves.Add("South");
	 }
	 if (!Map.IsWall(x-1,y)) {
	    validMoves.Add("West");
	 }
	 if (validMoves.Count == 0) {
	    return "North"; // Hopeless. Might as well go North!
	 } else {
	    Random rand = new Random();
	    int whichMove = rand.Next(validMoves.Count);
	    return validMoves[whichMove];
	 }
   }
   
   public static void Main()
   {
    while (true) {
	    Map.Initialize();
	    Map.MakeMove(MakeMove());
	 }
   }
}
       

File: starters/java/MyTronBot.java
// MyTronBot.java
// Author: your name goes here

import java.util.*;

class MyTronBot {
    public static String MakeMove() {
	int x = Map.MyX();
	int y = Map.MyY();
	List<String> validMoves = new ArrayList<String>();
	if (!Map.IsWall(x,y-1)) {
	    validMoves.add("North");
	}
	if (!Map.IsWall(x+1,y)) {
	    validMoves.add("East");
	}
	if (!Map.IsWall(x,y+1)) {
	    validMoves.add("South");
	}
	if (!Map.IsWall(x-1,y)) {
	    validMoves.add("West");
	}
	if (validMoves.size() == 0) {
	    return "North"; // Hopeless. Might as well go North!
	} else {
	    Random rand = new Random();
	    int whichMove = rand.nextInt(validMoves.size());
	    return validMoves.get(whichMove);
	}	
    }

    // Ignore this method. It's just doing boring stuff like communicating
    // with the contest tournament engine.
    public static void main(String[] args) {
	while (true) {
	    Map.Initialize();
	    Map.MakeMove(MakeMove());
	}
    }
}
